docs/guides/testing.md

```diff
@@ -197,31 +225,94 @@ In the next examples we are going to use `useCounterStore`
 > **Note**: all of these examples are written using TypeScript.
 
 ```ts
-// stores/user-counter-store.ts
-import { create } from 'zustand'
+// stores/counter-store-creator.ts
+import { type StateCreator } from 'zustand'
 
 export type CounterStore = {
   count: number
   inc: () => void
 }
 
-export const useCounterStore = create<CounterStore>()((set) => ({
+export const counterStoreCreator: StateCreator<CounterStore> = (set) => ({
   count: 1,
   inc: () => set((state) => ({ count: state.count + 1 })),
-}))
+})
+```
+
+```ts
+// stores/user-counter-store.ts
+import { create } from 'zustand'
+
+import { type CounterStore, counterStoreCreator } from './counter-store-creator'
+
+export const useCounterStore = create<CounterStore>()(counterStoreCreator)
+```
+
+```tsx
+// stores/use-counter-store-context.tsx
+import {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from 'react'
+import { type StoreApi, createStore } from 'zustand'
+import { useStoreWithEqualityFn } from 'zustand/traditional'
+import { shallow } from 'zustand/shallow'
+
+import { type CounterStore, counterStoreCreator } from './counter-store-creator'
+
+export const createCounterStore = () => {
+  return createStore<CounterStore>(counterStoreCreator)
+}
+
+export const CounterStoreContext = createContext<StoreApi<CounterStore>>(
+  null as never
+)
+
+export type CounterStoreProviderProps = PropsWithChildren
+
+export const CounterStoreProvider = ({
+  children,
+}: CounterStoreProviderProps) => {
+  const counterStoreRef = useRef(createCounterStore())
+
+  return (
+    <CounterStoreContext.Provider value={counterStoreRef.current}>
+      {children}
+    </CounterStoreContext.Provider>
+  )
+}
+
+export type UseCounterStoreContextSelector<T> = (store: CounterStore) => T
+
+export const useCounterStoreContext = <T,>(
```

Comment:
What does the comma do in generic type parameters?

Maybe a quick comment will be helpful if it's important?

Reply:
Doesn't do anything but for some reason prettier add it

Reply:
It seems that Prettier must add the trailing comma for the generic type list to be valid in TSX: https://github.com/prettier/prettier/issues/6114

It did it because this code snippet is `tsx`.